src/PlacePicker/PlacePicker.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import type { Place as PlaceType } from './PlacePickerTypes';
+import { StyleSheet } from '../PlatformStyleSheet';
+import { TagsInput } from '../TagsInput';
+import PlacePickerList from './components/PlacePickerList';
+
+type Props = {
+  onChangeText: (value: string) => void,
+  suggestedPlaces: PlaceType[],
+  onPressAdd?: (place: Array<PlaceType>) => void,
+  onPressItem?: (place: Array<PlaceType>) => void,
+  onRemoveSelectedPlace?: () => void,
+};
+
+type State = {
+  selected: Array<PlaceType>,
+  value: string,
+};
+
+type Event = { nativeEvent: { key: string } };
+
+export const filterSuggestedPlaces = (
+  suggestedPlaces: PlaceType[],
+  selectedPlaces: PlaceType[]
+) => {
+  const selectedIDs = selectedPlaces.map(place => place.id);
+  return suggestedPlaces.reduce((acc = [], place) => {
+    if (selectedIDs.includes(place.id)) {
+      return acc;
+    }
+    if (place.departurePoints) {
+      const filteredPoints = place.departurePoints.reduce((results, point) => {
+        if (selectedIDs.includes(point.id)) {
+          return results;
+        }
+        return [...results, point];
+      }, []);
+
+      return [...acc, { ...place, departurePoints: filteredPoints }];
+    }
+    return [...acc, place];
+  }, []);
+};
+
+export default class PlacePicker extends React.Component<Props, State> {
+  state = {
+    selected: [],
+    value: '',
+  };
+
+  handlePressAdd = (place: PlaceType) => {
+    const { onPressItem } = this.props;
+    this.setState(
+      ({ selected }) => ({ selected: [...selected, place] }),
+      () => {
+        const { selected } = this.state;
+        onPressItem?.(selected);
+      }
+    );
+  };
+
+  handlePressItem = (place: PlaceType) => {
+    const { onPressAdd } = this.props;
+    this.setState({ selected: [place] }, () => {
```

Comment:
So here you are completely overriding the `selected` state with `place`, when in the previous method you just append `place`. Is that the expected behaviour?

Reply:
Yes, it is expected behavior, when user click on `onPressItem` it should select just one place and redirect to the search form.

If user clicks on the `onPressAdd` it should add the city to selection, as a new Tag, update search context but the user won't be redirected back.